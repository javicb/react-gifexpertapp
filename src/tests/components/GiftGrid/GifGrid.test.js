import React from 'react';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../../hooks/useFetchGifs';
const { shallow } = require("enzyme")
const { GifGrid } = require("../../../components/GifGrid/GifGrid")
jest.mock('../../../hooks/useFetchGifs');

describe('Pruebas en el GifGrid', () => {
   const category = 'One Punch';

   test('Debe mostrarse correctamente ', () => {
      useFetchGifs.mockReturnValue({
         data: [],
         loading: true
      });

      const wrapper = shallow(<GifGrid category={category}></GifGrid>);
      expect(wrapper).toMatchSnapshot();
   });

   test('Debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
      const gifs = [{
         id: 'ABC',
         url: 'https://localhost/cualquergif',
         title: 'Cualquer cosa'
      }];

      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: false
      });

      const wrapper = shallow(<GifGrid category={category}></GifGrid>);
      expect(wrapper).toMatchSnapshot();
   });
})
