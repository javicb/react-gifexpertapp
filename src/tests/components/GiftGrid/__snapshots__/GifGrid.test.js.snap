// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGrid Debe mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquer cosa"
      url="https://localhost/cualquergif"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquer cosa"
      url="https://localhost/cualquergif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el GifGrid Debe mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
